win32/devdoc/socket_transport_win32.md

```diff
@@ -0,0 +1,296 @@
+﻿# socket transport Win32
+
+## Overview
+
+socket_transport_win32 is the module that abstracts the socket transport for the windows platform
+
+The `SOCKET_TYPE` determines which API are applicable for this instance of socket_transport.  The following chart shows which API's are allowed under which type:
+
+| API                                    | Allowable Socket Type
+|----------------------------------------|----------------------
+| socket_transport_create                | BOTH
+| socket_transport_destroy               | BOTH
+| socket_transport_connect               | SOCKET_CLIENT
+| socket_transport_listen                | SOCKET_SERVER
+| socket_transport_disconnect            | BOTH
+| socket_transport_accept                | SOCKET_SERVER
+| socket_transport_send                  | BOTH
+| socket_transport_receive               | BOTH
+| socket_transport_get_underlying_socket | BOTH
+
+## Exposed API
+
+```c
+typedef struct SOCKET_TRANSPORT_TAG* SOCKET_TRANSPORT_HANDLE;
+
+#define SOCKET_SEND_RESULT_VALUES \
+    SOCKET_SEND_OK, \
+    SOCKET_SEND_ERROR, \
+    SOCKET_SEND_SHUTDOWN
+
+MU_DEFINE_ENUM(SOCKET_SEND_RESULT, SOCKET_SEND_RESULT_VALUES)
+
+#define SOCKET_RECEIVE_RESULT_VALUES \
+    SOCKET_RECEIVE_OK, \
+    SOCKET_RECEIVE_WOULD_BLOCK, \
+    SOCKET_RECEIVE_ERROR, \
+    SOCKET_RECEIVE_SHUTDOWN
+
+MU_DEFINE_ENUM(SOCKET_RECEIVE_RESULT, SOCKET_RECEIVE_RESULT_VALUES)
+
+#define SOCKET_TYPE_VALUES \
+    SOCKET_CLIENT, \
+    SOCKET_SERVER
+
+MU_DEFINE_ENUM(SOCKET_TYPE, SOCKET_TYPE_VALUES)
+
+typedef struct SOCKET_BUFFER_TAG
```

Comment:
Interesting, no CONSTBUFFER here? I guess that makes sense when you don't need the reference counting in this layer

Reply:
And CONSTBUFFER is in c-util and not in PAL.  I thought about moving it, but like you said no ref counting.